src/dbnode/persist/fs/types.go

```diff
@@ -489,3 +491,45 @@ type BlockRetrieverOptions interface {
 	// IdentifierPool returns the identifierPool
 	IdentifierPool() ident.Pool
 }
+
+// ForEachRemainingFn is the function that is run on each of the remaining
+// series of the merge target that did not intersect with the fileset.
+type ForEachRemainingFn func(seriesID ident.ID, tags ident.Tags) error
```

Comment:
So related to my previous comment, this type would also accept a `[]xio.BlockReader`

Reply:
Ah I see how this would simplify things.